platform.md

```diff
@@ -146,94 +139,70 @@ The following is a non-exhaustive list of terms defined in the [OCI Distribution
+- The build image config and the run image config both specify the same `os`, `architecture`, `variant` (if specified), `io.buildpacks.distribution.name` (if specified), and `io.buildpacks.distribution.version` (if specified).
 
-A platform MAY support any number of mixins for a given stack in order to support application code or buildpacks that require those mixins.
+The platform SHOULD ensure that:
 
-Changes introduced by mixins SHOULD be restricted to the addition of operating system software packages that are regularly patched with strictly backwards-compatible security fixes.
-However, mixins MAY consist of any changes that follow the [Compatibility Guarantees](#compatibility-guarantees).
+- The image config's `variant` field is set to a valid identifier as defined in the [OCI Image Specification](https://github.com/opencontainers/image-spec/blob/main/config.md).
+- The image config's `Label` field has the label `io.buildpacks.distribution.name` set to the OS distribution and the label `io.buildpacks.distribution.version` set to the OS distribution version.
+  - For Linux-based images, each label should contain the values specified in `/etc/os-release` (`$ID` and `$VERSION_ID`), as the `os.version` field in an image config may contain combined distribution and version information.
+  - For Windows-based images, `io.buildpacks.distribution.name` should be empty; `io.buildpacks.distribution.version` should contain the value of `os.version` in the image config (e.g., `10.0.14393.1066`).
 
 ### Compatibility Guarantees
 
-Stack image authors SHOULD ensure that build image versions maintain [ABI-compatibility](https://en.wikipedia.org/wiki/Application_binary_interface) with previous versions, although violating this requirement will not change the behavior of previously built images containing app and launch layers.
-
-Stack image authors MUST ensure that new run image versions maintain [ABI-compatibility](https://en.wikipedia.org/wiki/Application_binary_interface) with previous versions.
-Stack image authors MUST ensure that app and launch layers do not change behavior when the run image layers are upgraded to newer versions, unless those behavior changes are intended to fix security vulnerabilities.
-
-Mixin authors MUST ensure that mixins do not affect the [ABI-compatibility](https://en.wikipedia.org/wiki/Application_binary_interface) of any object code compiled to run on the base stack images without mixins.
-
-During build, platforms MUST use the same set of mixins for the run image as were used in the build image (excluding mixins that have a stage specifier).
+- Base image authors SHOULD ensure that build image versions maintain [ABI-compatibility](https://en.wikipedia.org/wiki/Application_binary_interface) with previous versions, although violating this requirement will not change the behavior of previously built images containing app and launch layers.
```

Comment:
Is there a reason this is `SHOULD` vs. `MUST`. I'm just trying to think through the implications. If the ABI compatibility changes, but it's not known to the Buildpack Author, how will they know if they need to invalidate the buildpack cache? In the past they could use the `CNB_STACK_ID` for this purpose and store it as layer metadata.

Reply:
@samj1912 do you have an opinion here?

Reply:
I feel mixed about this - would the stack ID have always changed previously if there were changes to ABI compat? We seem to have relied on mixins for this purpose.

Reply:
@natalieparellano I'm not sure I understand what you're saying. Are you saying we only demanded ABI compatibility for mixins but not in the build? I'm looking at the [spec compat guarantees](https://github.com/buildpacks/spec/blob/main/platform.md#compatibility-guarantees) where this phrasing comes from:

> Stack image authors SHOULD ensure that build image versions maintain [ABI-compatibility](https://en.wikipedia.org/wiki/Application_binary_interface) with previous versions, although violating this requirement will not change the behavior of previously built images containing app and launch layers.

What you have written seems to match that, but I do wonder about the use case on where we want to break ABI compatibility here.

Reply:
> I feel mixed about this - would the stack ID have always changed previously if there were changes to ABI compat? We seem to have relied on mixins for this purpose.

That's fair. I think at Heroku, we've only cared about forward compatible changes in the stack for both the build + run image.

I do think it's worth discussing why build/run are different in the wording.

Reply:
My impression (could easily be wrong):

- it was previously a “should” because mixins were a workaround
- but nobody really used mixins, so in practice it was more like a “really ought to because you’ll have other problems if you don’t”
- dockerfile extensions are still a potential workaround/backdoor that could create issues by creating an ABI incompatibility that isn’t reflected in the target ID.
- it probably doesn’t cost anyone anything to change this to a MUST, and it’s probably good advice / good practice to follow.

@natalieparellano et. al. wdyt about changing this to a MUST ?

Reply:
> why build/run are different in the wording

I think that's because of rebase. I guess in theory, if you change ABI compat for the build image, you can always also upgrade your buildpacks. 

I'm happy to change SHOULD -> MUST here if that's desired.

Reply:
> who is "you" in that case

You're right. In my head, it was the "platform operator" who is presumably overseeing builder creation. It is fair to point out that the personas may be different depending on the organization.

Reply:
I've made #368 to move the discussion to...